crates/core_arch/src/x86/avx512f.rs

```diff
@@ -16156,6 +16156,126 @@ pub unsafe fn _mm_maskz_compress_pd(k: __mmask8, a: __m128d) -> __m128d {
     transmute(vcompresspd128(a.as_f64x2(), _mm_setzero_pd().as_f64x2(), k))
 }
 
+/// Contiguously store the active 32-bit integers in a (those with their respective bit set in writemask k) to unaligned memory at base_addr.
+///
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm512_mask_compressstoreu_epi32)
+#[inline]
+#[target_feature(enable = "avx512f")]
+#[cfg_attr(test, assert_instr(vpcompressd))]
+pub unsafe fn _mm512_mask_compressstoreu_epi32(base_addr: *mut i8, k: __mmask16, a: __m512i) {
```

Comment:
Intel's intrinsic guide uses a `void*` for `base_addr`, the llvm intrinsics use an `i8*`. Using a ptr of the correct datatype would be more ergonomic, but I'm not sure whether that might prevent using the intrinsics for actually unaligned data.

Reply:
The convention here is to use `*mut u8` where C uses void pointers. LLVM's `i8` doesn't mean anything since LLVM IR types don't have signs: LLVM's `i8` is used for both of Rust's `u8` and `i8`.

Reply:
I missed that convention when implementing the masked load/store instructions, there are also several more intrinsics that did not already follow this convention. I can take a look at adjusting the `stdarch-verify` test to catch this and change any existing type differences. Since avx512 is still unstable that should be possible I guess.